### PR TITLE
Da#137 add support for listing indexes

### PR DESCRIPTION
**Pull request by AayushTyagi1**
On expanding the Scope we now get 2 folders:
- Collections
- Indexes
Collections: Same behavior as old
Indexes: List all indexes for the target scope.

  1. Each index should use the "name" of the index as the file's name, and the extension should be "*.n1ql"
  2. Once the user clicks on the index, a new tab should open with the index definition(e.g. `CREATE INDEX `def_city` ON `travel-sample`(`city`) WITH {  "defer_build":true }`). If it is capella we will show a JSON document containing index information.
  3. Users must be able to "refresh" the indexes folder
 
Screenshot 2023-04-19 at 3 01 26 PM

### CODE REVIEW

**Comment by AayushTyagi1** [comment]
Please don't merge. I am going for several QA rounds before merging this PR.

File: src/extension.ts
Comment (biozal) [inline src/extension.ts:100]: Should this be a utility function that you ask for the name of, and it figures out how to split it apart from the pathParts?  This just seems like a log of magic ints that are looking at indexes without understanding why you need part 3 instead of part 2.  At least we should have a breakdown of what this is indexes-wise (what is 0, 1, 2, 3, and 4).  This would make maintaining this code a lot easier than guesswork.

Comment (AayushTyagi1) [reply src/extension.ts:100]: This has been used at many places in code so I made it to a single functionality function so that if indexes change I can just refactor this function. We are storing the path from bucket to a particular document. 
I will put the comments in code and signify what's there in part 3.

**Review by deniswsrosa — CHANGES_REQUESTED** [review]
Looks good but it does require some documentation

**Review by deniswsrosa — APPROVED** [review]

**Pull request merged**
